sympy/polys/multivariate_resultants.py

```diff
@@ -0,0 +1,357 @@
+from sympy.polys.monomials import monomial_deg
+from sympy.polys.monomials import itermonomials
+from sympy.polys.orderings import monomial_key
+from sympy.polys.polytools import poly_from_expr
+from sympy.functions.combinatorial.factorials import binomial
+
+class DixonResultant():
+    """
+    A class for retrieving the Dixon's resultant of a multivariate system.
+
+    Examples
+    ========
+    >>> from sympy.core import symbols
+
+    >>> from sympy.polys.multivariate_resultants import DixonResultant
+    >>> x, y = symbols('x, y')
+
+    >>> p = x + y
+    >>> q = x ** 2 + y ** 3
+    >>> h = x ** 2 + y
+
+    >>> dixon = DixonResultant(variables=[x, y], polynomials=[p, q, h])
+    >>> poly = dixon.get_dixon_polynomial()
+    >>> matrix = dixon.get_dixon_matrix(polynomial=poly)
+    >>> matrix
+    Matrix([
+    [ 0,  0, -1,  0, -1],
+    [ 0, -1,  0, -1,  0],
+    [-1,  0,  1,  0,  0],
+    [ 0, -1,  0,  0,  1],
+    [-1,  0,  0,  1,  0]])
+    >>> matrix.det()
+    0
+
+    Reference
+    ==========
+    1. [Kapur1994]_
+    2. [Palancz08]_
+
+    See Also
+    ========
+    Notebook in examples: sympy/example/notebooks.
+    """
+
+    def __init__(self, polynomials, variables):
+        """
+        A class that takes two lists, a list of polynomials and list of
+        variables. Returns the Dixon matrix of the multivariate system.
+
+        Parameters
+        ----------
+        polynomials : list of polynomials
+            A  list of m n-degree polynomials
+        variables: list
+            A list of all n variables
+        """
+        self.polynomials = polynomials
+        self.variables = variables
+
+        self.n = len(self.variables)
+        self.m = len(self.polynomials)
+
+        a = IndexedBase("alpha")
+        # A list of n alpha variables (the replacing variables)
+        self.dummy_variables = [a[i] for i in range(self.n)]
+
+        # A list of the d_max of each variable.
+        self.max_degrees = [poly_from_expr(f, self.variables)[0].degree() for f in self.polynomials]
+
+    def get_dixon_polynomial(self):
+        r"""
+        Returns
+        -------
+
+        dixon_polynomial: polynomial
+            Dixon's polynomial is calculated as:
+
+            delta = Delta(A) / ((x_1 - a_1) ... (x_n - a_n)) where,
+
+            A =  |p_1(x_1,... x_n), ..., p_n(x_1,... x_n)|
+                 |p_1(a_1,... x_n), ..., p_n(a_1,... x_n)|
+                 |...             , ...,              ...|
+                 |p_1(a_1,... a_n), ..., p_n(a_1,... a_n)|
+        """
+        if self.m != (self.n + 1):
+            raise ValueError('Method invalid for given combination.')
+
+        # First row
+        rows = [self.polynomials]
+
+        temp = list(self.variables)
+        iterator = iter(self.dummy_variables)
+
+        for idx in range(self.n):
+            temp[idx] = next(iterator)
+            substitution = {self.variables[i]: temp[i] for i in range(len(temp))}
```

Comment:
Could this be something like `substitution = {var: t for var, t in zip(self.variables, temp)}`?